unicodeString toTrimmedUTF8 panics with newline characters in the middle

We found this by giving `parseInt` a not only ascii character string with a newline in the middle. So for example `нещо\nтук` panics parseInt. 

The reason behind this is that [the regex](https://github.com/dop251/goja/blob/6438c8ddc5177eb6d3679f7c363b6bc84a4c37c4/string_unicode.go#L34) tries to match from the beginning of the string to the end and `.` by [default doesn't match `\n`](https://github.com/google/re2/wiki/Syntax)(Sorry, can't link directly to the flags).

This could be fixed with adding the flag `s` in front as `(?s)^...` the rest of the regex, but I propose we actually use [strings.Trim](https://golang.org/pkg/strings/#Trim) -  `strings.Trim(s.String(), parser.WhitespaceChars)` instead  of this regex. strings.TrimSpace is used for the [asciiString](https://github.com/dop251/goja/blob/6438c8ddc5177eb6d3679f7c363b6bc84a4c37c4/string_ascii.go#L303), so I don't know why regex was used here? 

This is from the initial commit so I guess this comes from otto?

Also to be honest I am not particularly certain why goja does utf8<->utf16 conversation in the one case and not the other. I would argue that it will probably be more performant to do it only when it is actually needed for something(whatever that is), but I guess this is for another discussion.

stack trace from [k6](https://github.com/loadimpact/k6) as a proof:
```
panic: runtime error: index out of range [1] with length 0 [recovered]
        panic: runtime error: index out of range [1] with length 0 [recovered]
        panic: runtime error: index out of range [1] with length 0

goroutine 72 [running]:
github.com/loadimpact/k6/vendor/github.com/dop251/goja.AssertFunction.func1.1(0xc001325d20)
        /home/mstoykov/.gvm/pkgsets/go1.14.1/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/runtime.go:1485 +0x98
panic(0xe0f1a0, 0xc00210b700)
        /home/mstoykov/.gvm/gos/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*vm).try.func1(0xc0034e6000, 0x0, 0xc001325bf8, 0x0, 0x0, 0x0, 0xc001325c70)
        /home/mstoykov/.gvm/pkgsets/go1.14.1/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/vm.go:377 +0x3da
panic(0xe0f1a0, 0xc00210b700)
        /home/mstoykov/.gvm/gos/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/loadimpact/k6/vendor/github.com/dop251/goja.unicodeString.toTrimmedUTF8(0xc000e71818, 0x3, 0x3, 0x3, 0xc001ab9560)
        /home/mstoykov/.gvm/pkgsets/go1.14.1/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/string_unicode.go:103 +0x1aa
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*Runtime).builtin_parseInt(0xc0034e4000, 0x10e22e0, 0x19d49e0, 0xc0020530d0, 0x2, 0x31, 0x10e2060, 0x19d49f2)
        /home/mstoykov/.gvm/pkgsets/go1.14.1/global/src/github.com/loadimpact/k6/vendor/github.com/dop251/goja/builtin_global.go:29 +0x8f
github.com/loadimpact/k6/vendor/github.com/dop251/goja.(*vm)._nativeCall(0xc0034e6000, 0xc0034f3550, 0x2)
```

--- a/string_unicode.go
+++ b/string_unicode.go
@@ -9,7 +9,6 @@ import (
 	"io"
 	"math"
 	"reflect"
-	"regexp"
 	"strings"
 	"unicode/utf16"
 	"unicode/utf8"
@@ -28,10 +27,6 @@ type runeReaderReplace struct {
 
 var (
 	InvalidRuneError = errors.New("Invalid rune")
-)
-
-var (
-	unicodeTrimRegexp = regexp.MustCompile("^[" + parser.WhitespaceChars + "]*(.*?)[" + parser.WhitespaceChars + "]*$")
 )
 
 func (rr runeReaderReplace) ReadRune() (r rune, size int, err error) {
@@ -96,7 +91,7 @@ func (s unicodeString) toTrimmedUTF8() string {
 	if len(s) == 0 {
 		return ""
 	}
-	return unicodeTrimRegexp.FindStringSubmatch(s.String())[1]
+	return strings.Trim(s.String(), parser.WhitespaceChars)
 }
 
 func (s unicodeString) ToNumber() Value {

